### PR TITLE
Allow Home to get manifest from NPM

### PR DESCRIPTION
**Pull request by frankrousseau**
If there is a package field in the app descriptor (from Cozy registry), it fetches additional metadata from NPM registry instead of the app git repository.

### CODE REVIEW

File: server/lib/manifest.coffee [hunk 1]
```diff
@@ -17,22 +42,36 @@ class exports.Manifest
...
-            logger.warn 'App manifest without git URL'
+            logger.warn(
+                'App manifest without recognized git URL or package field')
             logger.raw app
-            callback null
+            callback null, {}
+
+
+    # Download the manifest from the NPM registry. This manifest has many
+    # information (it includes history). So we extract the lastest version
```
Comment (nono) [inline server/lib/manifest.coffee]: s/lastest/latest/

File: server/lib/manifest.coffee [hunk 2]
```diff
@@ -17,22 +42,36 @@ class exports.Manifest
...
+    # of the application manifest.
+    downloadFromNpm: (packageName, callback) ->
+        client = request.createClient "https://registry.npmjs.org/"
+        client.get packageName, (err, res, data) ->
+            manifest = data.versions[data['dist-tags'].latest]
```
Comment (nono) [inline server/lib/manifest.coffee]: It should be guarded by `unless err`

File: server/lib/manifest.coffee [hunk 3]
```diff
@@ -17,22 +42,36 @@ class exports.Manifest
...
+            if err? and res.statusCode is 404
```
Comment (nono) [inline server/lib/manifest.coffee]: Maybe `res?.statusCode` (I think you can have an `err` and no `res` in some cases)

File: server/lib/manifest.coffee (hunk 1 above)
Comment (frankrousseau) [inline server/lib/manifest.coffee]: I think it's latest.

Comment (frankrousseau) [inline server/lib/manifest.coffee]: https://github.com/sindresorhus/latest-version

Comment (nono) [inline server/lib/manifest.coffee]: Yes, but you wrote `lastest` in the comment

Comment (frankrousseau) [inline server/lib/manifest.coffee]: My bad sorry.

File: server/lib/manifest.coffee [hunk 4]
```diff
@@ -17,22 +42,39 @@ class exports.Manifest
...
+            callback null, {}
+
+
+    # Download the manifest from the NPM registry. This manifest has many
+    # information (it includes history). So we extract the latest version
+    # of the application manifest.
+    downloadFromNpm: (packageName, callback) ->
+        client = request.createClient "https://registry.npmjs.org/"
+        client.get packageName, (err, res, data) ->
+            if res.statusCode is 404
```
Comment (nono) [inline server/lib/manifest.coffee]: `res?.statusCode`. I think it's possible to have an `err` but no `res` (DNS resolution failure for example).

Comment (frankrousseau) [inline server/lib/manifest.coffee]: Yep.

**Comment by nono** [comment]
Thanks!

**Pull request merged**